operator/internal/handlers/lokistack_create_or_update.go

```diff
@@ -26,17 +26,23 @@ import (
 )
 
 // CreateOrUpdateLokiStack handles LokiStack create and update events.
-func CreateOrUpdateLokiStack(ctx context.Context, req ctrl.Request, k k8s.Client, s *runtime.Scheme, flags manifests.FeatureFlags) error {
+func CreateOrUpdateLokiStack(
+	ctx context.Context,
+	req ctrl.Request,
+	k k8s.Client,
+	s *runtime.Scheme,
+	flags manifests.FeatureFlags,
+) (*status.DegradedCondition, error) {
```

Comment:
Something seems strange - for lack of a better word - about returning the optional condition next to an error. It like they both signify that something has gone wrong. I wonder if it might be better to pass in the `secrets` rather than extract it in this method.

Reply:
I don't see that as a good option. We should extract secrets here itself and pass the errors out of it.
DegradedCondition seems to be optional condition but I don't see any other naming for doing this. I will leave it this way for the time being.

Reply:
@Red-GV we had thought about a few alternatives to the current approach. One of them was to define a custom error type that contains the information about the degraded state, but in the end we settled on just returning an additional value.

Does the custom error type sound like a better alternative to you?

Reply:
Yeah @xperimental, a custom error type is probably the better option in my opinion. The degraded condition already acts like its own error. So checking two things when they both say "An error has occurred which stopped the creation or updated of the objects" is kinda odd/clunky to me.

Reply:
Sometimes I just need to write the actual code to get a feel for it. Have a look: [comparison with PR](https://github.com/sasagarw/loki/compare/update-statuses...xperimental:degraded-as-err)

Reply:
Thanks @xperimental :+1: